nav2_controller/plugins/test/progress_checker.cpp

```diff
@@ -0,0 +1,135 @@
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <memory>
+#include <string>
+
+#include "gtest/gtest.h"
+#include "nav2_controller/plugins/simple_progress_checker.hpp"
+#include "nav_2d_utils/conversions.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+
+using nav2_controller::SimpleProgressChecker;
+
+class TestLifecycleNode : public nav2_util::LifecycleNode
+{
+public:
+  explicit TestLifecycleNode(const std::string & name)
+  : nav2_util::LifecycleNode(name)
+  {
+  }
+
+  nav2_util::CallbackReturn on_configure(const rclcpp_lifecycle::State &)
+  {
+    return nav2_util::CallbackReturn::SUCCESS;
+  }
+
+  nav2_util::CallbackReturn on_activate(const rclcpp_lifecycle::State &)
+  {
+    return nav2_util::CallbackReturn::SUCCESS;
+  }
+
+  nav2_util::CallbackReturn on_deactivate(const rclcpp_lifecycle::State &)
+  {
+    return nav2_util::CallbackReturn::SUCCESS;
+  }
+
+  nav2_util::CallbackReturn on_cleanup(const rclcpp_lifecycle::State &)
+  {
+    return nav2_util::CallbackReturn::SUCCESS;
+  }
+
+  nav2_util::CallbackReturn onShutdown(const rclcpp_lifecycle::State &)
+  {
+    return nav2_util::CallbackReturn::SUCCESS;
+  }
+
+  nav2_util::CallbackReturn onError(const rclcpp_lifecycle::State &)
+  {
+    return nav2_util::CallbackReturn::SUCCESS;
+  }
+};
+
+void checkMacro(
+  nav2_core::ProgressChecker & pc,
+  double x0, double y0,
+  double x1, double y1,
+  int delay,
+  bool expected_result)
+{
+  pc.reset();
+  geometry_msgs::msg::PoseStamped pose0, pose1;
+  pose0.pose.position.x = x0;
+  pose0.pose.position.y = y0;
+  pose1.pose.position.x = x1;
+  pose1.pose.position.y = y1;
+  EXPECT_TRUE(pc.check(pose0));
+  rclcpp::sleep_for(std::chrono::seconds(delay));
+  if (expected_result) {
+    EXPECT_TRUE(pc.check(pose1));
+  } else {
+    EXPECT_FALSE(pc.check(pose1));
+  }
+}
+
+TEST(SimpleProgressChecker, progress_checker_reset)
+{
+  auto x = std::make_shared<TestLifecycleNode>("progress_checker");
+
+  nav2_core::ProgressChecker * pc = new SimpleProgressChecker;
+  pc->reset();
+  delete pc;
+  EXPECT_TRUE(true);
```

Comment:
I don't understand how the reset method is being tested. Also, do we need `EXPECT_TRUE(true);`?

Reply:
Making sure it doesn't crash as the next test is dependent on it. EXPECT_TRUE is used as a convention for returning from TEST functions. I have based these tests on the already available goal_checker tests.

Reply:
Ah alright, thanks for explaining it:)